### PR TITLE
Retry until router deploys in e2e test for metrics

### PR DESCRIPTION
**Pull request by smarterclayton**
The metrics test flaked out because the router was still deploying when the e2e test ran (it was the first test). We don't block provision exit on router deploying yet.

### CODE REVIEW

**Comment by smarterclayton** [comment]
Fixes a test flake, labelling

File: test/extended/router/metrics.go [hunk 1]
```diff
@@ -58,13 +58,17 @@ var _ = g.Describe("[Conformance][Area:Networking][Feature:Router] openshift rou
 			}
 		}
 
-		epts, err := oc.AdminKubeClient().CoreV1().Endpoints("default").Get("router", metav1.GetOptions{})
+		// wait for the router endpoints to show up
+		err = wait.PollImmediate(2*time.Second, 120*time.Second, func() (bool, error) {
+			epts, err := oc.AdminKubeClient().CoreV1().Endpoints("default").Get("router", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
```
Comment (kargakis) [inline test/extended/router/metrics.go]: Don't you want to retry if the endpoint does not exist yet?

Comment (smarterclayton) [reply test/extended/router/metrics.go]: Odds of that happening are so low as to be not worth it.  If the endpoints controller has taken 2-3 minutes to create the endpoint we're broke.

Comment (kargakis) [reply test/extended/router/metrics.go]: Or the deployer pod has yet to start.

Comment (smarterclayton) [reply test/extended/router/metrics.go]: No, endpoints object is created immediately.  It's just empty.

**Pull request merged**